mayastor/tests/nexus_rebuild.rs

```diff
@@ -226,15 +244,273 @@ async fn nexus_test_child(child: u64) {
+                .expect_err("rebuild job not created yet");
+        }
+        let src = RebuildJob::lookup(&get_dev(children))
+            .expect("now the job should exist")
+            .source
+            .clone();
+
+        for child in 0 .. children {
+            if get_dev(child) != src {
+                RebuildJob::lookup_src(&get_dev(child))
+                    .iter()
+                    .filter(|s| s.destination != get_dev(child))
+                    .inspect(|&job| {
+                        log::error!(
+                            "Job {:?} should be associated with src child {}",
+                            job,
+                            child
+                        );
+                    })
+                    .any(|_| panic!("Should not have found any jobs!"));
+            }
+        }
+
+        assert_eq!(
+            RebuildJob::lookup_src(&src)
+                .iter()
+                .inspect(|&job| {
+                    assert_eq!(job.destination, get_dev(children));
+                })
+                .count(),
+            1
+        );
+        nexus
+            .add_child(&get_dev(children + 1), false)
+            .await
+            .unwrap();
+        let _ = nexus.start_rebuild(&get_dev(children + 1)).unwrap();
+        assert_eq!(RebuildJob::lookup_src(&src).len(), 2);
+
+        nexus.remove_child(&get_dev(children)).await.unwrap();
+        nexus.remove_child(&get_dev(children + 1)).await.unwrap();
+        nexus_lookup(nexus_name()).unwrap().destroy().await.unwrap();
+    });
+
+    test_fini();
+}
+
+#[test]
+// todo: decide whether to keep the idempotence on the operations or to
+// create a RPC version which achieves the idempotence
+fn rebuild_operations() {
+    test_ini("rebuild_operations");
+
+    Reactor::block_on(async {
+        nexus_create(1).await;
+        let nexus = nexus_lookup(nexus_name()).unwrap();
+
+        nexus
+            .resume_rebuild(&get_dev(1))
+            .await
+            .expect_err("no rebuild to resume");
+
+        nexus_add_child(1, false).await;
+
+        nexus
+            .resume_rebuild(&get_dev(1))
+            .await
+            .expect("already running");
+
+        nexus.pause_rebuild(&get_dev(1)).await.unwrap();
+        reactor_poll!(10);
+        // already pausing so no problem
+        nexus.pause_rebuild(&get_dev(1)).await.unwrap();
+
+        let _ = nexus
+            .start_rebuild(&get_dev(1))
+            .expect_err("a rebuild already exists");
+
+        nexus.stop_rebuild(&get_dev(1)).await.unwrap();
+        common::wait_for_rebuild(
+            get_dev(1),
+            RebuildState::Stopped,
+            // already stopping, should be enough
+            std::time::Duration::from_millis(250),
+        )
+        .unwrap();
+        // already stopped
+        nexus.stop_rebuild(&get_dev(1)).await.unwrap();
+
+        nexus_lookup(nexus_name()).unwrap().destroy().await.unwrap();
+    });
+
+    test_fini();
+}
+
+#[test]
```

Comment:
can you add a comment to spell out what it means to rebuild concurent? it not obvious from looking at the code what is happening here.

Reply:
yep will do.
(it basically adds N children to the nexus causing N rebuild jobs to be created; it then removes the children and tries to add them back again to make sure new rebuild jobs get created)

Reply:
I've updated the fn name (concurrently was perhaps an unfortunate choice) and comments plus the linter fixes